asf_search/download/download.py

```diff
@@ -74,14 +73,21 @@ def download_url(url: str, path: str, filename: str = None, session: ASFSession
         for chunk in response.iter_content(chunk_size=8192):
             f.write(chunk)
 
-def remotezip(url: str, session: ASFSession) -> RemoteZip:
+def remotezip(url: str, session: ASFSession):
     """
     :param url: the url to the zip product
     :param session: the authenticated ASFSession to read and download from the zip file
     """
-
-    session.hooks['response'].append(strip_auth_if_aws)
-    return RemoteZip(url, session=session)
+    try:
+        import remotezip
+    except ImportError:
+        remotezip = None
+    
+    if remotezip is None:
+        ASF_LOGGER.warn("Could not find remotezip package in current python environment. \"remotezip\" is an optional dependency of asf-search required for the `remotezip()` method, and can be install via `python3 -m pip install asf-search[extras]`")
```

Comment:
```suggestion
    if RemoteZip is None:
        raise ImportError("Could not find remotezip package in current python environment. \"remotezip\" is an optional dependency of asf-search required for the `remotezip()` method, and can be install via `python3 -m pip install asf-search[extras]`")
```

A couple of thoughts here:
1. The way Python package discovery works, you'll typically have to exit out of the interpreter to be able to see a newly installed package anyways, and this is esp. true in Jupyter (need to restart the Kernel) so I would raise here instead of delaying this check and warning. If you raise, you can move the try-except import block up to the top so it's only evaluated once.

2. Moving it to the top would allow you to keep the inport form the same as before:
   ```python
       try:
        from  remotezip import RemoteZip
    except ImportError:
        RemoteZip = None
   ```
   (since ModuleNotFoundError is a subclass of ImportError it'll work this way and keep the normal optional import form)
   
   Which would allow you to keep via a forward declaration the type hint (I added a suggestion for this change) and avoids potential namespace clashes